<robot name="pliers">

  <material name="white">
    <color rgba="0 1 0 1"/>
  </material>

  <link name="base_link" />

  <link name="pliers_link" />
  <joint name="pliers_joint" type="fixed">
    <parent link="base_link"/>
    <child link="pliers_link"/>
    <origin xyz="1 1 0" />
  </joint>


  <link name="left_pliers_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/barry/catkin_ws/src/tool_meshes/meshes/pliers_left.dae"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <joint name="left_joint" type="revolute">
    <parent link="pliers_link"/>
    <child link="left_pliers_link"/>
    <axis xyz="0 0 1" />
    <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
  </joint>

  <link name="right_pliers_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/barry/catkin_ws/src/tool_meshes/meshes/pliers_right.dae"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="right_joint" type="revolute">
    <parent link="pliers_link"/>
    <child link="right_pliers_link"/>
    <axis xyz="0 0 -1" />
    <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
  </joint>


  <link name="clamp_link" >
    <visual>
      <geometry>
        <mesh filename="file:///home/barry/catkin_ws/src/tool_meshes/meshes/clamp_body.dae"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <joint name="clamp_joint" type="fixed">
    <parent link="base_link"/>
    <child link="clamp_link"/>
    <origin xyz="1 0 0" />
  </joint>


  <link name="clamp_handle_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/barry/catkin_ws/src/tool_meshes/meshes/clamp_handle.dae"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="clamp_handle_joint" type="prismatic">
    <parent link="clamp_link"/>
    <child link="clamp_handle_link"/>
    <axis xyz="1 0 0" />
    <limit effort="1000.0" lower="0.1" upper="0.30" velocity="0.5"/>
  </joint>

</robot>
